File: csharp/072_scrabble-score/ScrabbleScore.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ScrabbleScore
{
    private static readonly Dictionary<char, int> LetterPoints = CreateLetterPoints();
    private static Dictionary<char, int> CreateLetterPoints()
    {
        var map = new Dictionary<char, int>();
        AddLetters(map, 1, "AEIOULNRST");
        AddLetters(map, 2, "DG");
        AddLetters(map, 3, "BCMP");
        AddLetters(map, 4, "FHVWY");
        AddLetters(map, 5, "K");
        AddLetters(map, 8, "JX");
        AddLetters(map, 10, "QZ");
        return map;
    }
    private static void AddLetters(Dictionary<char, int> map, int points, string letters)
    {
        foreach (char letter in letters) 
            map[letter] = points;
    }
    public static int Score(string input)
        => input.ToUpper().Sum(c => LetterPoints.GetValueOrDefault(c, 0));
}

File: javascript/ex_055_scrabble-score/scrabble-score.js
const mapa = [[1,/[AEIOULNRST]/gi], [2,/[DG]/gi], [3,/[BCMP]/gi], [4,/[FHVWY]/gi], [5,/K/gi], [8,/[JX]/gi], [10,/[QZ]/gi]];
export const score = word => mapa.reduce((count, [points, re]) => (word.match(re)?.length ?? 0) * points + count, 0);
